### PR TITLE
support features indexer with length 0

### PR DESCRIPTION
**Pull request by corochann**

### CODE REVIEW

**Comment by corochann** [comment]
Could you review @mottodora

File: chainer_chemistry/dataset/indexer.py [hunk 1]
```diff
@@ -159,16 +159,24 @@ def _extract_feature(self, data_index, j):
...
-
-            res = [self.extract_feature(i, j) for i in data_index]
+            if len(data_index) == 0:
+                try:
+                    # HACKING
+                    return self.extract_feature_by_slice(slice(0, 0, 1), j)
+                except ExtractBySliceNotSupportedError:
+                    res = []
+            else:
+                if isinstance(data_index[0], (bool, numpy.bool, numpy.bool_)):
```
Comment (mottodora) [inline chainer_chemistry/dataset/indexer.py]: This is not related to this PR. Maybe we need to consider when data_index[0] is numpy.int.

**Comment by mottodora** [comment]
LGTM
